### PR TITLE
Add AMI build and release

### PR DESCRIPTION
**Pull request by dchw**
Builds an Amazon Linux 2 based AMI for each new earthly release.

### CODE REVIEW

File: release/release.sh [hunk 1]
```diff
@@ -115,3 +118,10 @@ if [ "$release_apt_and_yum" = "true" ]; then
 else
     echo "WARNING: there is no staging environment for apt or yum repos"
 fi
+
+if [ "$release_ami" = "true" ]; then
+  "$earthly" --push --build-arg RELEASE_TAG ./ami+update-pipelines
+  "$earthly" --push --build-arg RELEASE_TAG ./ami+build-ami
```
Comment (vladaionescu) [inline release/release.sh:124]: The `--build-arg` syntax is deprecated.

```suggestion
  "$earthly" --push ./ami+update-pipelines --RELEASE_TAG="$RELEASE_TAG"
  "$earthly" --push ./ami+build-ami --RELEASE_TAG="$RELEASE_TAG"
```

**Review by vladaionescu — APPROVED** [review]

File: release/release.sh (hunk 1 above)
Comment (dchw) [reply release/release.sh:124]: Yeah I was matching the rest of the script using the old style still.

File: release/ami/Earthfile [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        --mount=type=secret,id=+secrets/user/earthly-technologies/aws/config,target=/root/.aws/config \
+
+        terraform init && \
+        terraform workspace select $ENVIRONMENT
+
+    RUN \
+        --mount=type=secret,id=+secrets/user/earthly-technologies/aws/credentials,target=/root/.aws/credentials \
+        --mount=type=secret,id=+secrets/user/earthly-technologies/aws/config,target=/root/.aws/config \
+
+        terraform plan
```
Comment (alexcb) [inline release/ami/Earthfile:46]: nit: can this be moved up to the previous RUN command?

```
        terraform init && \
        terraform workspace select $ENVIRONMENT && \
        terraform plan
```

File: release/ami/imagebuilder.tf [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  infrastructure_configuration_arn = aws_imagebuilder_infrastructure_configuration.earthly.arn
+  distribution_configuration_arn   = aws_imagebuilder_distribution_configuration.earthly[each.key].arn
+  name                             = "earthly-${replace(each.key, ".", "_")}"
+  description                      = "Builds an AMI with Earthly ${each.key}"
+}
+
+output "pipelines" {
+  value       = aws_imagebuilder_image_pipeline.earthly
+  description = "The pipelines we might build; used so we can trigger them manually from another Earthfile."
+}
```
Comment (alexcb) [inline release/ami/imagebuilder.tf:114]: nit: missing newline.

**Review by alexcb — APPROVED** [review]

**Review by alexcb — APPROVED** [review]

**Pull request merged**